numerics/matrix_computations_body.hpp

```diff
@@ -20,6 +20,72 @@ using quantities::Abs;
 using quantities::Pow;
 using quantities::Sqrt;
 
+// This is J(p, q, θ) in [GV13] section 8.5.1.  This matrix is also called a
+// Givens rotation.  As mentioned in [GV13] section 5.1.9, "It is critical that
+// the special structure of a Givens rotation matrix be exploited".
+struct JacobiRotation {
+  double cos;
+  double sin;
+  int p;
+  int q;
+};
+
+// See [GV13] section 8.5.2, algorithm 8.5.1.
+template<typename Scalar, typename Matrix>
+JacobiRotation SymmetricShurDecomposition2(Matrix const& A,
+                                           int const p,
+                                           int const q) {
+  DCHECK_LE(0, p);
+  DCHECK_LT(p, q);
+  DCHECK_LT(q, A.rows());
+  static Scalar const zero{};
```

Comment:
`constexpr Scalar zero{}`, let’s not have locking to initialize a local static zero.